src/Evolution/Systems/NewtonianEuler/Limiters/Weno.hpp

```diff
@@ -0,0 +1,205 @@
+      typename ConservativeVarsWeno::package_argument_tags;
+
+  /// \brief Package data for sending to neighbor elements
+  void package_data(
+      gsl::not_null<PackagedData*> packaged_data,
+      const Scalar<DataVector>& mass_density_cons,
+      const tnsr::I<DataVector, VolumeDim>& momentum_density,
+      const Scalar<DataVector>& energy_density, const Mesh<VolumeDim>& mesh,
+      const std::array<double, VolumeDim>& element_size,
+      const OrientationMap<VolumeDim>& orientation_map) const noexcept;
+
+  using limit_tags =
+      tmpl::list<NewtonianEuler::Tags::MassDensityCons,
+                 NewtonianEuler::Tags::MomentumDensity<VolumeDim>,
+                 NewtonianEuler::Tags::EnergyDensity>;
+  using limit_argument_tags =
+      tmpl::list<domain::Tags::Mesh<VolumeDim>,
+                 domain::Tags::Element<VolumeDim>,
+                 domain::Tags::SizeOfElement<VolumeDim>,
+                 domain::Tags::DetInvJacobian<Frame::Logical, Frame::Inertial>,
+                 evolution::dg::Tags::NormalCovectorAndMagnitude<VolumeDim>,
+                 ::hydro::Tags::EquationOfStateBase>;
+
+  /// \brief Limit the solution on the element
+  template <size_t ThermodynamicDim>
+  bool operator()(
+      gsl::not_null<Scalar<DataVector>*> mass_density_cons,
+      gsl::not_null<tnsr::I<DataVector, VolumeDim>*> momentum_density,
+      gsl::not_null<Scalar<DataVector>*> energy_density,
+      const Mesh<VolumeDim>& mesh, const Element<VolumeDim>& element,
+      const std::array<double, VolumeDim>& element_size,
+      const Scalar<DataVector>& det_inv_logical_to_inertial_jacobian,
+      const typename evolution::dg::Tags::NormalCovectorAndMagnitude<
+          VolumeDim>::type& normals_and_magnitudes,
+      const EquationsOfState::EquationOfState<false, ThermodynamicDim>&
+          equation_of_state,
+      const std::unordered_map<
+          std::pair<Direction<VolumeDim>, ElementId<VolumeDim>>, PackagedData,
+          boost::hash<std::pair<Direction<VolumeDim>, ElementId<VolumeDim>>>>&
+          neighbor_data) const noexcept;
+
+ private:
+  template <size_t LocalDim>
+  // NOLINTNEXTLINE(readability-redundant-declaration) false positive
+  friend bool operator==(const Weno<LocalDim>& lhs,
+                         const Weno<LocalDim>& rhs) noexcept;
+
+  ::Limiters::WenoType weno_type_;
+  NewtonianEuler::Limiters::VariablesToLimit vars_to_limit_;
+  double neighbor_linear_weight_;
+  std::optional<double> tvb_constant_;
+  std::optional<double> kxrcf_constant_;
+  bool apply_flattener_;
+  bool disable_for_debugging_;
+  // Note: conservative_vars_weno_ is always used for calls to package_data, and
+  // is also used when limiting cons vars with the simple WENO algorithm. So we
+  // construct conservative_vars_weno_ with the correct TVB constant when it is
+  // used for limiting (precisely, when weno_type_ == SimpleWeno), and with a
+  // dummy TVB constant value of 0.0 otherwise (weno_type_ == Hweno). This lets
```

Comment:
would setting to `signaling_NaN()` instead of `0.0` work?

Reply:
It does 👍